File: spendy_backend/src/main/java/com/rohitcodes/expense_tracker/service/DashboardService.java
package com.rohitcodes.expense_tracker.service;

import com.rohitcodes.expense_tracker.dto.dashboard.SummaryResponse;
import com.rohitcodes.expense_tracker.dto.transaction.TransactionResponse;

import java.util.List;

public interface DashboardService {

    SummaryResponse getMonthlySummary(Long userId, Integer year, Integer month);

    List<TransactionResponse> getRecentTransactions(Long userId);
}

File: spendy-mobileFrontend/src/services/dashboardService.ts
import { api } from "@/src/services/api";
import type { SummaryResponse } from "@/src/types/dashboard";
import type { Transaction } from "@/src/types/transaction";

export const dashboardService = {
  async getSummary(userId: number, year: number, month: number) {
    const response = await api.get<SummaryResponse>(`/users/${userId}/dashboard/summary`, {
      params: { year, month }
    });
    return response.data;
  },

  async getRecentTransactions(userId: number) {
    const response = await api.get<Transaction[]>(`/users/${userId}/dashboard/recent-transactions`);
    return response.data;
  }
};